holoviews/plotting/plotly/renderer.py

```diff
@@ -24,11 +24,141 @@
   }});
 }});
 var plotly = window._Plotly || window.Plotly;
+
+// Restore axis range extents set by the user
+for (var key in data.layout) {{
+    if(key.slice(1, 5) === 'axis') {{
+        if (plot.layout[key] && plot.layout[key].range) {{
+            data.layout[key].range = plot.layout[key].range;
+        }}
+    }}
+}}
+
 plotly.relayout(plot, data.layout);
 plotly.redraw(plot);
 """
 
 
+def build_new_plot_js(plot_id,
+                      data,
+                      layout,
+                      config,
+                      comm_ids,
+                      timeout,
+                      debounce,
+                      callback_handlers):
+
+    # Based on JS_CALLBACK from pyviz_comms
+    return """
+var plotdiv = document.getElementById('{plot_id}');
+var plotly = window._Plotly || window.Plotly;
+plotly.newPlot(plotdiv, {data}, {layout}, {config}).then(function() {{
+    var elem = document.getElementById("{plot_id}.loading");
+    elem.parentNode.removeChild(elem);
+
+    function unique_events(events) {{
```

Comment:
Could we just get most of this from ``pyviz_comms.JS_CALLBACK``?

Reply:
We could if we refactor `pyviz_comms.JS_CALLBACK` a little. The utility functions are all the same, but this requires the initialization of a list of comms (`JS_CALLBACK` is for just 1) and we don't want to logic to construct `data` here.

Reply:
Would you like me to try refactoring `pyviz_comms` a bit to reuse parts of `JS_CALLBACK` here?  That's fine with me, I just wasn't sure how tightly you want to link versions of `holoviews` and `pyviz_comms` (i.e. can `holoviews` require the newest version of `pyviz_comms` or does there need to be some fallback for older versions?).